generic3g/tests/Test_VerticalLinearMap.pf

```diff
@@ -0,0 +1,47 @@
+#include "MAPL_TestErr.h"
+
+module Test_VerticalLinearMap
+
+   use mapl3g_CSR_SparseMatrix, only: SparseMatrix_sp => CSR_SparseMatrix_sp
+   use mapl3g_VerticalLinearMap, only: compute_linear_map_fixedlevels_to_fixedlevels
+   use mapl3g_VerticalLinearMap, only: apply_linear_map
+   use funit
+   use, intrinsic :: iso_fortran_env, only: REAL32
+
+   implicit none
+
+contains
+
+   @test
+   subroutine test_linear_map_fixedlevels_to_fixedlevels()
+
+      real(REAL32), allocatable :: vcoord_src(:), vcoord_dst(:)
+      real(REAL32), allocatable :: fin(:), fout(:)
+      ! real(REAL32), allocatable :: matrix(:, :)
+      type(SparseMatrix_sp) :: matrix
+      integer :: status
+
+      vcoord_src = [30., 20., 10.]
+      vcoord_dst = [20., 10.]
+      call compute_linear_map_fixedlevels_to_fixedlevels(vcoord_src, vcoord_dst, matrix, _RC)
+      fin = [7., 8., 3.]
+      call apply_linear_map(matrix, fin, fout)
+      @assertEqual(fout, [8., 3.])
```

Comment:
At the very least the assertion order is wrong.  For `assertEqual` it is (expected, found).  But `@assert_that` is preferable when usable as it is here.   And it's order is (found, expected)
```suggestion
      @assert_that(fout, is([8.,3.]))
```

Reply:
With `@assert_that(fout, is([8., 3.]))`, I get the error
```
error #6053: Structure constructor may not have components with the PRIVATE attribute
  call assert_that(fout, is([8., 3.]), &
-----------------------------^
```
so switched to `@assertEqual([8., 3.], fout)`, and that works.

Reply:
Yeah - there probably is a way to use `all()` or something, but not worth any more trouble for now.

Reply:
ok